### PR TITLE
Accept uniqueness-token option to add to HTTP_HEADERS

### PR DESCRIPTION
**Pull request by sethtrain**
Recent changes to the NationBuilder API allow for the use of a uniqueness token to be sent along with a request.  If a subsequent request with the same uniqueness token if received within 2 hours, it will be rejected with a `409 Conflict`

cc: @kaip @codealchemy

### CODE REVIEW

File: lib/nationbuilder/client.rb [hunk 1]
```diff
@@ -39,13 +39,14 @@ def base_url
...
-  def raw_call(path, method, body = {}, args = {})
+  def raw_call(path, method, body = {}, args = {}, opts = {})
     url = NationBuilder::URL.new(base_url).generate_url(path, args)
 
     request_args = {
       header: {
         'Accept' => 'application/json',
-        'Content-Type' => 'application/json'
+        'Content-Type' => 'application/json',
+        'X-Request-ID' => opts[:uniqueness_token]
```
Comment (kaip) [inline lib/nationbuilder/client.rb]: What happens if the uniqueness_token is not set?

Comment (sethtrain) [inline lib/nationbuilder/client.rb]: I've confirmed in the console that HTTPClient.send won't error with a header of nil:

`HTTPClient.send(:get, "http://google.com", {header: {'X-Request-ID' => nil}})`

Currently trying to ensure it won't cause the branch to succeed if `nil`

**Comment by kaip** [comment]
:+1: LGTM

**Comment by kaip** [comment]
@codealchemy Do we need to add any documentation about this change?

**Comment by codealchemy** [comment]
Yes, I think it would be helpful to add something to the readme for using a uniqueness token.

**Comment by sethtrain** [comment]
@kaip @codealchemy 

I've confirmed that X-Request-ID will be sent if nil as an empty string so I refactored the headers out into a function that only adds X-Request-ID when it is actually needed.

**Comment by kaip** [comment]
:+1:

**Comment by sethtrain** [comment]
@codealchemy have a peek at the README in the last commit and let me know if that is in line with your desire.

**Comment by DavidHuie** [comment]
How does the NationBuilder API treat empty `X-Request-ID` headers? If it's pedantic, all nationbuilder-rb requests that ignore the `uniqueness_token` parameter in `#call` should draw the 409 because the token is the same, albeit empty.

**Comment by sethtrain** [comment]
@DavidHuie The NationBuilder API uses the header information to cache the request id and not allow anything to process with that given ID for the next 2 hours.  I think the API is the place for that to sit since we have the infrastructure to control that globally and don't have to keep that in the client code.  But you definitely have a point about empty `X-Request-ID`.  I'm not handling the case when it is blank,  I think if it is blank or nil we don't want the API to care about it.  So I am going to fix that.

**Comment by codealchemy** [comment]
@sethtrain :+1: for the readme update

**Comment by kaip** [comment]
:+1:

**Comment by codealchemy** [comment]
:rocket:

**Pull request merged**